web-animations-2/Overview.bs

```diff
@@ -3043,6 +3094,41 @@ with which it is associated is updated.
 
 </div>
 
+<div class="members">
+:   <dfn dict-member for=KeyframeAnimationOptions>id</dfn>
+::  The string to assign to the generated {{Animation}}'s
+    {{Animation/id}} attribute.
+
+:   <dfn dict-member for=KeyframeAnimationOptions>timeline</dfn>
+::  An optional value which, if present,
+    specifies the <a>timeline</a> with which to associate
+    the newly-created <a>animation</a>.
+
+:   <dfn dict-member for=KeyframeAnimationOptions>rangeStart</dfn>
+::  An optional value which, if present,
+    specifies the start of the <a>animation</a>’s [=timeline attachment range=].
+    A {{DOMString}} value is parsed as an 'animation-range-end' value
+    to yield a {{TimelineRangeOffset}} {{TimelineRangeOffset/rangeName}}
+    and {{TimelineRangeOffset/offset}};
+    a {{CSSNumericValue}} is interpreted as a {{TimelineRangeOffset}}
+    with that {{TimelineRangeOffset/offset}} and a null {{TimelineRangeOffset/rangeName}}.
+    The {{TimelineRangeOffset/rangeName}} and {{TimelineRangeOffset/offset}}
+    are the interpreted as defined for 'animation-range-end'.
```

Comment:
I don't follow what this sentence means (or really this paragraph). What does interpreted mean?

Reply:
It means follow the definitions in https://drafts.csswg.org/scroll-animations-1/#animation-range-start for turning the rangeName and offset into a point on the timeline. I didn't want to inline those instructions, because they're changing as we adjust the syntax, defaulting rules, and capabilities of `animation-range`.

Reply:
For better or worse this spec is pretty procedural, especially from "Programming interface" onwards, so that you can mostly just convert spec prose into implementation code. I'm just wondering what an implementer is supposed to do with "interpreted as".

From the point of view of this interface, as long as `rangeName` and `rangeOffset` get set to the right values, I think we're done. How those values are applied goes somewhere else (ideally somewhere in Web Animations which might then refer to definitions in scroll animations).

Reply:
> How those values are applied goes somewhere else (ideally somewhere in Web Animations which might then refer to definitions in scroll animations).

Well, yeah, but that needs to be defined _somewhere_, and we don't have the conceptual infrastructure yet to reference it here. I can move the interpretation out of the definition here, and put it in a separate paragraph, but I can't update Web Animations 2 to define the attachment range concepts without a significant amount of additional editing work in sections that have not yet been ported over.

Reply:
Thanks, yeah, I think we should keep the description here just to how the value gets set. If we don't yet have anywhere to park the description of how those values are applied, we could add it here as a note of some sort indicating that it eventually needs to live somewhere else.